syncano/models/custom_sockets.py

```diff
@@ -0,0 +1,160 @@
+            dependencies=self.dependencies_data
+        )
+
+        created_socket._raw_data['links'] = created_socket._raw_data['links'].links_dict
+        self.to_python(created_socket._raw_data)
+        return self
+
+    def update(self):
+        if self.is_new():
+            raise SyncanoValueError('Install socket first.')
+
+        update_socket = self.__class__.please.update(
+            name=self.name,
+            endpoints=self.endpoints_data,
+            dependencies=self.dependencies_data
+        )
+
+        update_socket._raw_data['links'] = update_socket._raw_data['links'].links_dict
+        self.to_python(update_socket._raw_data)
+        return self
+
+    def recheck(self):
+        recheck_path = self.links.recheck
+        connection = self._get_connection()
+        rechecked_socket = connection.request('POST', recheck_path)
+        self.to_python(rechecked_socket)
+        return self
+
+
+class SocketEndpoint(Model):
+    """
+    OO wrapper around endpoints defined in CustomSocket instance.
+    Look at the custom socket documentation for more details.
+
+    :ivar name: :class:`~syncano.models.fields.StringField`
+    :ivar calls: :class:`~syncano.models.fields.JSONField`
+    :ivar links: :class:`~syncano.models.fields.LinksField`
+    """
+    name = fields.StringField(max_length=64, primary_key=True)
+    allowed_methods = fields.JSONField()
+    links = fields.LinksField()
+
+    class Meta:
+        parent = CustomSocket
+        endpoints = {
+            'detail': {
+                'methods': ['get'],
+                'path': '/endpoints/{name}/'
+            },
+            'list': {
+                'methods': ['get'],
+                'path': '/endpoints/'
+            }
+        }
+
+    def run(self, method='GET', data=None):
+        endpoint_path = self.links.self
+        connection = self._get_connection()
+        if not self._validate_method(method):
+            raise SyncanoValueError('Method: {} not specified in calls for this custom socket.'.format(method))
+        method = method.lower()
+        if method in ['get', 'delete']:
+            response = connection.request(method, endpoint_path)
+        elif method in ['post', 'put', 'patch']:
+            response = connection.request(method, endpoint_path, data=data or {})
+        else:
+            raise SyncanoValueError('Method: {} not supported.'.format(method))
+        return response
+
+    @classmethod
+    def get_all_endpoints(cls, instance_name=None):
+        connection = cls._meta.connection
+        all_endpoints_path = Instance._meta.resolve_endpoint(
+            'endpoints',
+            {'name': cls.please.properties.get('instance_name') or instance_name}
+        )
+        response = connection.request('GET', all_endpoints_path)
+        return [cls(**endpoint) for endpoint in response['objects']]
+
+    def _validate_method(self, method):
+        if '*' in self.allowed_methods or method in self.allowed_methods:
```

Comment:
why not `return '*' in self.allowed_methods or method in self.allowed_methods` ?